bazel-mode.el

```diff
@@ -0,0 +1,240 @@
+
+(setq glob-args
+  '("exclude" "exclude_directories" "include"))
+
+(setq go-repositories-args
+  '("go_version"))
+
+(setq go-repository-args
+  '("build_file_generation" "build_file_name" "build_tags" "importpath"
+	"go-prefix" "tag" "vcs"))
+
+(setq http-args
+  '("executable" "strip_prefix" "type" "url" "urls"))
+
+(setq java-buildrule-args
+  '("classpath_resources" "create_executable" "deploy_manifest_lines"
+	"javacopts" "jvm_flags" "launcher" "main_class" "output_licenses" "plugins"
+	"resources" "resource_jars" "resource_strip_prefix" "restricted_to"
+	"runtime_deps" "use_testrunner"))
+
+(setq local-repository-args
+  '("path"))
+
+(setq maven-args
+  '("artifact" "repository" "server" "sha1"))
+
+(setq new-repository-args
+  '("build_file" "build_file_content" "workspace_file" "workspace_file_content"))
+
+(setq package-args
+  '("default_deprecation" "default_testonly" "default_visibility"))
+
+(setq py-buildrule-args
+  '("default_python_version" "default_runtime" "files" "imports" "interpreter"
+    "interpreter_path" "main" "protoc" "srcs_version"))
+
+(setq bazel-types
+  (sort-strings-longest-first
+    (append bind-args cc-buildrule-args common-buildrule-args common-testrule-args
+	  config-setting-args genrule-args genquery-args git-repository-args glob-args
+	  go-repositories-args go-repository-args http-args java-buildrule-args
+	  local-repository-args maven-args new-repository-args package-args
+	  py-buildrule-args)))
+
+(setq bazel-constants
+  '("true"))
+
+(setq bazel-events
+  '())
+
+;; generate regex string for each category of keywords
+(setq bazel-keywords-regexp (regexp-opt bazel-keywords 'words))
+(setq bazel-functions-regexp (regexp-opt bazel-functions 'words))
+(setq bazel-type-regexp (regexp-opt bazel-types 'words))
+(setq bazel-constant-regexp (regexp-opt bazel-constants 'words))
+(setq bazel-event-regexp (regexp-opt bazel-events 'words))
+
+;; create the list for font-lock.
+;; each category of keyword is given a particular face
+(setq bazel-font-lock-keywords
+      `(
+        (,bazel-keywords-regexp . font-lock-keyword-face)
+        (,bazel-functions-regexp . font-lock-function-name-face)
+        (,bazel-type-regexp . font-lock-type-face)
+        (,bazel-constant-regexp . font-lock-constant-face)
+        (,bazel-event-regexp . font-lock-builtin-face)
+        ;; note: order above matters, because once colored, that part won't change.
+        ;; in general, longer words first
+        ))
+
+(setq bazel-mode-syntax-table
+  (let ((table (make-syntax-table)))
+	;; comments start with '#' and end with line break
+	(modify-syntax-entry ?# "<" table)
+	(modify-syntax-entry ?\n ">" table)
+	table))
+
+;;;###autoload
+(define-derived-mode bazel-mode prog-mode "Bazel"
+  "Major mode for editing Bazel BUILD and WORKSPACE files"
+
+  ;; code for syntax highlighting
+  (setq font-lock-defaults '((bazel-font-lock-keywords)))
+
+  (set-syntax-table bazel-mode-syntax-table))
```

Comment:
It isn't necessary to set the syntax table explicitly, `define-derived-mode` already does that: https://www.gnu.org/software/emacs/manual/html_node/elisp/Derived-Modes.html

Reply:
Removed.